packages/abstraction/test/SchemaValidator.test.ts

```diff
@@ -0,0 +1,73 @@
+import path from 'path';
+import { validateSchema } from "../src/SchemaValidator";
+import { ContractGenerator } from "../src";
+import { MultipleContractAdapter, EOSContractAdapter, ICONContractAdapter, ModelContractAdapter } from "@aloxide/bridge";
+
+describe('test SchemaValidator', () => {
+
+  let contractGenerator;
+  const logger = {
+    info: jest.fn(),
+    debug: jest.fn(),
+    error: jest.fn(),
+    success: jest.fn(),
+  };
+  const adapter = new MultipleContractAdapter();
+
+  adapter.addAdapters(
+    new EOSContractAdapter(),
+    new ICONContractAdapter(),
+    new ModelContractAdapter(),
+  );
+
+  const checkName = val => {
+    return /^[1-5a-zA-Z]+/.test(val)
+  }
+
+  const checkType = val => {
+    const supportedType = ["uint64_t", "number", "string", "array", "bool"];
+    return supportedType.indexOf(val) !== -1
+  }
+
+  const requiredSchema = {
+    entities: [{
+      name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+      fields: [{
+        name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+        type: { type: String, required: true, use: { checkType } },
+      }],
+      key: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } }
+    }
+    ]
+  };
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
+    contractGenerator = new ContractGenerator({
+      aloxideConfigPath: path.resolve(__dirname, './aloxide.yml'),
+      resultPath: path.resolve(__dirname, './out'),
+      adapter,
+      logger
+    });
+  })
+  describe('test function validateExtraFields', () => {
+    it('should validateSchema successful', () => {
+      const schemaErrors = validateSchema(contractGenerator.aloxideConfig, requiredSchema, logger);
+      expect(schemaErrors.length).toBeLessThan(1);
+    });
+
+    it('should validateSchema failed', () => {
+      const schemaErrors = validateSchema(contractGenerator.aloxideConfig, null, logger);
+      expect(schemaErrors.length).toBeGreaterThanOrEqual(1);
```

Comment:
I think `expect(schemaErrors.length >= 1).toBeTrue();` is simpler.

Reply:
updated